torch/_library/autograd.py

```diff
@@ -102,9 +123,37 @@ def backward(ctx, *grads):
     ):
         Generated = supports_tensorlist(Generated)
 
+
+
+    Generated_jvp = type(
+        name + "_jvp",
+        (torch.autograd.function._SingleLevelFunction,),
+        {
+            "forward": staticmethod(forward_jvp),
+            "jvp": staticmethod(jvp),
+            "setup_context": staticmethod(setup_context_fn),
+        },
+    )
+
+    def jvp(*args):
+        result = forward_no_grad(*args)
+
     # The dispatcher passes any keyword-only-args as kwargs and the
     # rest of the args (even if specified as kwargs) as args.
     def autograd_impl(keyset, *args, **keyword_only_args):
+
+        interpreter = torch._C._functorch.peek_interpreter_stack()
+        if interpreter is not None:
+            interpreter = coerce_cinterpreter(interpreter)
+            if isinstance(interpreter, JvpInterpreter):
+                if not info._jvp_fn:
+                    raise RuntimeError(
+                        f"Trying to run jvp on {op} but no jvp "
+                        f"formula was registered. "
+                        f"Please use register_jvp to add one."
+                    )
+            result = custom_function_call_grad(interpreter, Generated_jvp, *args, **keyword_only_args)
+            return result
```

Comment:
We might not need this: functorch is able to re-use autograd (including jvp) registrations, it automatically puts some interpreter logic around it. So our approach should be to mirror how built-in PyTorch ops handle jvp. Because of that, it should be sufficient to just add a jvp staticmethod to the `Generated` autograd.Function (instead of making a new one), see the previous comment